wagtail/admin/views/scheduled_pages.py

```diff
@@ -0,0 +1,113 @@
+from datetime import timedelta
+
+from django.conf import settings
+from django.core.exceptions import PermissionDenied
+from django.shortcuts import get_object_or_404, redirect
+from django.template.response import TemplateResponse
+from django.utils import timezone
+from django.utils.http import url_has_allowed_host_and_scheme
+from django.utils.translation import gettext_lazy as _
+
+from wagtail.admin import messages
+from wagtail.admin.ui.components import Component
+from wagtail.models import Page
+from wagtail.permission_policies.pages import PagePermissionPolicy
+
+
+def get_scheduled_pages_for_user(request):
+    user_pages = PagePermissionPolicy().instances_user_has_permission_for(
+        request.user, "publish"
+    )
+    pages = (
+        Page.objects.annotate_approved_schedule()
+        .filter(_approved_schedule=True)
+        .prefetch_related("content_type")
+        .order_by("-first_published_at")
+        & user_pages
```

Comment:
My reading of this is that `user_pages` isn't being filtered by ones which are scheduled? Perhaps combining the querysets should be earlier. Or build a new one based on the ids of the 2.

Reply:
It perhaps makes more sense to do:

```python
user_pages.annotate_approved_schedule()
.filter(_approved_schedule=True)
.prefetch_related("content_type")
.order_by("-first_published_at")
```